olp-cpp-sdk-core/include/olp/core/utils/LruCache.h

```diff
@@ -59,48 +63,131 @@ class LruCache {
     inline const Value& value() const;
 
    protected:
+    /// A typename of the map constant iterator.
     typename MapType::const_iterator m_it;
   };
 
+  /// A constant iterator of the `LruCahce` object.
   class const_iterator : public ValueType {
    public:
+    /// A typedef for the iterator category.
     typedef std::bidirectional_iterator_tag iterator_category;
+    /// A typedef for the difference type.
     typedef std::ptrdiff_t difference_type;
+    /// A typedef for the `ValueType` type.
     typedef ValueType value_type;
+    /// A typedef for the `ValueType` constant reference.
     typedef const value_type& reference;
+    /// A typedef for the `ValueType` constant pointer.
     typedef const value_type* pointer;
 
+    /**
+     * @brief Creates a constant iterator object.
+     * 
+     */
     const_iterator() = default;
+    /**
+     * @brief Creates a constant iterator object.
+     * 
+     */
     const_iterator(const const_iterator&) = default;
+    /**
+     * @brief Copies this and the specified iterator to this.
+     * 
+     * @return A refenrence to this object.
+     */
     const_iterator& operator=(const const_iterator&) = default;
 
+    /**
+     * @brief Checks whether the values of the `const_iterator`
+     * parameter are the same as the values of the `other` parameter.
+     * 
+     * @param other The `const_iterator` instance.
+
+     * @return True if the values are the same; false otherwise.
+     */
     inline bool operator==(const const_iterator& other) const;
+    /**
+     * @brief Checks whether the values of the `const_iterator`
+     * parameter are not the same as the values of the `other` parameter.
+     * 
+     * @param other The `const_iterator` instance.
+     * 
+     * @return True if the values are not the same; false otherwise.
+     */
     inline bool operator!=(const const_iterator& other) const {
       return !operator==(other);
     }
 
+    /**
+     * @brief Iterates to the next `LruCache` object.
+     * 
+     * @return A reference to this.
+     */
     inline const_iterator& operator++();
+    /**
+     * @brief Iterates the specified number of times to the next `LruCache` object.
+     * 
+     * @return A new constant iterator.
+     */
     inline const_iterator operator++(int);
 
+    /**
+     * @brief Iterates to the previous `LruCache` object.
+     * 
+     * @return A reference to this.
+     */
     inline const_iterator& operator--();
+    /**
+     * @brief Iterates the specified number of times to the previous `LruCache` object.
+     * 
+     * @return A new constant iterator.
+     */
     inline const_iterator operator--(int);
 
+    /**
+     * @brief Gets a reference to this object.
+     * 
+     * @return The reference to this.
+     */
     reference operator*() const { return *this; }
+
+    /**
+     * @brief Gets a pointer to this object.
+     * 
+     * @return The pointer to this.
+     */
     pointer operator->() const { return this; }
```

Comment:
Usually the default operators do not need to be commented as they are obvious or they only have a simple `///` doxygen comment. But i guess if you already did the job we can leave as is.